test/OpenTelemetry.Tests/Logs/OpenTelemetryLoggerProviderTests.cs

```diff
@@ -84,4 +86,49 @@ public void VerifyOptionsCannotBeChangedAfterInit(bool initialValue)
         Assert.Equal(initialValue, provider.Options.IncludeScopes);
         Assert.Equal(initialValue, provider.Options.ParseStateValues);
     }
+
+    [Fact]
+    public void VerifyAddProcessorOverloadWithImplementationFactory()
```

Comment:
@Yun-Ting Some feedback...

* These tests don't have anything to do with OtlpExporter they are just verifying SDK functionality so we should probably move them into OpenTelemetry.Tests.

* Please don't use Mock. Ever. 🤣 Here is how you could rewrite this test to just use all the real bits:

```csharp
[Fact]
public void VerifyAddProcessorOverloadWithImplementationFactory()
{
   var services = new ServiceCollection();
   services.AddSingleton<MyProcessor>();
 
   services.AddLogging(logging => logging.AddOpenTelemetry(
      options => options.AddProcessor((sp, builder) => builder.AddProcessor(sp.GetRequiredService<MyProcessor>())));
 
   using var sp = services.BuildServiceProvider();
 
   var loggerProvider = sp.GetRequiredService<LoggerProvider>() as LoggerProviderSdk;
 
   Assert.NotNull(loggerProvider);
   Assert.NotNull(loggerProvider.Processor);
   Assert.True(loggerProvider.Processor is MyProcessor);
}
```

Reply:
I've addressed the feedback. Thank you!